csrank/metrics.py

```diff
@@ -116,24 +116,96 @@ def zero_one_rank_loss_for_scores_ties(y_true, s_pred):
 
 
 def make_ndcg_at_k_loss(k=5):
+    r"""Computes the Normalized Discounted Cumulative Gain
+
+    The Discounted Cumulative Gain is the sum of the document's relevancies,
+    logarithmically discounted by their rank. That means the DCG is higher when
+    the more relevant documents are highly ranked, lower otherwise.
+
+    Concretely:
+
+    .. math::
+        \mathrm{DCG}_p = \sum_{i = 1}^p \frac{\mathit{rel}_i}{\log_2(i + 1)}
+
+    Where :math:`\mathit{rel}_i` is the relevance of the document that is
+    ranked at :math:`i`. Since this library deals with ranks, not relevances,
+    it is necessary to define a conversion between the two. We define the
+    relevance of an item as :math:`2^{\mathit{inv}}` where :math:`\mathit{inv}`
+    is the negative rank normalized to :math:`[0, 1]`. An alternative way to
+    view this is that :math:`\mathit{inv}` is the relevancy and our definition
+    of ndcg exponentially discounts relevancies.
+
+    To make the DCG comparable across different ranking (particularly rankings
+    of different length), it is normalized by the ideal DCG. The resulting nDCG
+    can be described as
+
+    .. math::
+        \mathrm{nDCG}_p = \frac{\mathrm{DCG}_p}{\mathrm{IDCG}_p}
+
+    with
+
+    .. math::
+        \mathrm{IDCG}_p
+        = \sum_{i = 1}^{\lvert \mathit{REL}_p \rvert} \frac{mathit{rel}}{\log_2(i + 1)}
```

Comment:
`= \sum_{i = 1}^{\lvert \mathit{REL}_p \rvert} \frac{\mathit{rel}}{\log_2(i + 1)}`
there is a `\` missing in front of the `mathit`

Reply:
Right, good catch! I intended to build the docs once before clearing this for review, but I forgot. I've done that now.

Doing that I noticed that sphinx doesn't even build comprehensive api docs (just docs for a subset of the api). That's probably something we should change.